File: code_blocks/🧰 Test & Launch/errors_1.m
if (error) {

    // log error details
    NSLog(@"RCError: %@", [error.userInfo objectForKey:RCReadableErrorCodeKey]);
    NSLog(@"Message: %@", error.localizedDescription);
    NSLog(@"Underlying Error: %@", [error.userInfo objectForKey:NSUnderlyingErrorKey]);

    switch ([error code]) {
        case RCNetworkError:
            showError(@"Network error, check your connection and try again.");
        case RCPurchaseNotAllowedError:
            showError(@"Purchases not allowed on this device.");
        case RCPurchaseInvalidError:
            showError(@"Purchase invalid, check payment source.");
        default:
            break;
    }

}

File: code_blocks/🧰 Test - Launch/errors_1.swift
if let error = error as? RevenueCat.ErrorCode {
  print(error.errorCode)
  print(error.errorUserInfo)

  switch error {
  case .purchaseNotAllowedError:
    showAlert("Purchases not allowed on this device.")
  case .purchaseInvalidError:
    showAlert("Purchase invalid, check payment source.")
  default: break
  }
} else {
  // Error is a different type
}
